smarts/core/renderer.py

```diff
@@ -60,91 +47,115 @@ def required_to(cls, thing):
         )
 
 
-class _ShowBaseInstance(ShowBase):
-    """ Wraps a singleton instance of ShowBase from Panda3D. """
-
-    def __new__(cls):
-        # Singleton pattern:  ensure only 1 ShowBase instance
-        if "__it__" not in cls.__dict__:
-            # disable vsync otherwise we are limited to refresh-rate of screen
-            loadPrcFileData("", "sync-video false")
-            loadPrcFileData("", "model-path %s" % os.getcwd())
-            # TODO: the following speeds up rendering a bit... might consider it.
-            # loadPrcFileData("", "model-cache-dir %s/.panda3d_cache" % os.getcwd())
-            loadPrcFileData("", "audio-library-name null")
-            loadPrcFileData("", "gl-version 3 3")
-            loadPrcFileData("", "notify-level error")
-            loadPrcFileData("", "print-pipe-types false")
-            # https://www.panda3d.org/manual/?title=Multithreaded_Render_Pipeline
-            # loadPrcFileData('', 'threading-model Cull/Draw')
-            # have makeTextureBuffer create a visible window
-            # loadPrcFileData('', 'show-buffers true')
-        it = cls.__dict__.get("__it__")
-        if it is None:
-            cls.__it__ = it = object.__new__(cls)
-            it.init()
-        return it
-
-    def __init__(self):
-        pass  # singleton pattern, uses init() instead (don't call super().__init__() here!)
-
-    def init(self):
-        self._render_lock = Lock()
-        try:
-            # There can be only 1 ShowBase instance at a time.
-            super().__init__(windowType="offscreen")
+try:
+    import gltf
+    from direct.showbase.ShowBase import ShowBase
+    from panda3d.core import (
+        FrameBufferProperties,
+        GraphicsOutput,
+        GraphicsPipe,
+        NodePath,
+        OrthographicLens,
+        Shader,
+        Texture,
+        WindowProperties,
+        loadPrcFileData,
+    )
```

Comment:
I could suggest leaving a comment on why we are wrapping this in the try block.

Reply:
Sure.